src/tkn.ts

```diff
@@ -692,29 +682,21 @@ export class TknImpl implements Tkn {
     const eventListenerNode = new TektonNodeImpl(TknImpl.ROOT, 'EventListener', ContextType.EVENTLISTENERNODE, this, TreeItemCollapsibleState.Collapsed);
     const conditionsNode = new TektonNodeImpl(TknImpl.ROOT, 'Conditions', ContextType.CONDITIONSNODE, this, TreeItemCollapsibleState.Collapsed);
     pipelineTree.push(pipelineNode, pipelineRunNode, taskNode, clustertaskNode, taskRunNode, pipelineResourceNode, triggerTemplatesNode, triggerBindingNode, eventListenerNode, conditionsNode);
-    this.cache.set(pipelineNode, await this.getPipelines(pipelineNode));
-    this.cache.set(pipelineRunNode, await this.getPipelineRunsList(pipelineRunNode));
-    this.cache.set(taskNode, await this.getTasks(taskNode));
-    this.cache.set(clustertaskNode, await this.getClusterTasks(clustertaskNode));
-    this.cache.set(taskRunNode, await this.getTaskRunList(taskRunNode));
-    this.cache.set(pipelineResourceNode, await this.getPipelineResources(pipelineResourceNode));
-    this.cache.set(triggerTemplatesNode, await this.getTriggerTemplates(triggerTemplatesNode));
-    this.cache.set(triggerBindingNode, await this.getTriggerBinding(eventListenerNode));
-    this.cache.set(eventListenerNode, await this.getEventListener(eventListenerNode));
-    this.cache.set(conditionsNode, await this.getConditions(conditionsNode));
     return pipelineTree;
-
   }
 
-  async refreshPipelineRun(pipelineName: string): Promise<void> {
-    await kubectl.watchPipelineRun(pipelineName);
-    pipelineExplorer.refresh();
+  async refreshPipelineRun(pipeline: TektonNode): Promise<void> {
+    await kubectl.watchPipelineRun(pipeline.getName(), () => {
+      pipelineExplorer.refresh(pipeline);
```

Comment:
<img width="434" alt="Screenshot 2020-04-17 at 1 53 42 PM" src="https://user-images.githubusercontent.com/9924513/79548525-3979b400-80b3-11ea-86b6-5632610d4153.png">

When pipelineRun finish its execution it doesn't refresh the separate tree of pipelineRun and TaskRun